projects/common/lib/services/page-state.service.ts

```diff
@@ -0,0 +1,107 @@
+import { Injectable } from '@angular/core';
+import { Route } from '@angular/router';
+
+export interface PageList {
+  index: number;
+  path: string;
+  isComplete: boolean;
+  [key: string]: any;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PageStateService {
+
+  pageList: PageList[] = [];
+
+  constructor() { }
+
+  setPages( arr: Route[],
+            routeListConst: any,
+            fn?: (x: any) => {[key: string]: any} ): PageList[] {
+
+    if ( this.pageList && !this.pageList.length ) {
+
+      const routeConst = Object.keys( routeListConst ).map( x => routeListConst[x] );
+
+      let cnt = 0;
+      this.pageList = arr.filter((itm: any) => !itm.redirectTo)
+        .map((itm: any) => {
+          const val = routeConst.find( x => x.path === itm.path );
+
+          cnt = cnt + 1; // count for indices
+
+          // Page List to be returned
+          let obj = {
+            index: cnt,
```

Comment:
Assuming you name the variable "index", you can just use this syntactic sugar below:

```suggestion
            index,  
```